### DISCUSSION TITLE
Preserving prefetched staleTime and other options during hydration

### DISCUSSION THREAD

**Discussion by goszczynskip** [TanStack/query#2903] [Q&A] [answered]
I'm trying to understand how to make prefetching work with Next.js SSR. I'm prefetching data in the `getStaticProps` and later hydrating page as described in the docs. Queries are hydrated correctly but options are missing. If you'll look into react-query dev tools staleTime is missing. This results in queries fetching on initial load. 

Here is sandbox: https://codesandbox.io/s/stoic-dhawan-sbx41?file=/pages/index.js

I'm wondering how can I make react-query keep data fresh even though it was fetched in getStaticProps. Dehydrate function seems to ignore query options.

**Comment by TkDodo** [comment, marked as answer]
all options are not being dehydrated - only the queries / mutations are. What you need to do is set the staleTime on the client so that it doesn't refetch. Here's a working fork of your sandbox: https://codesandbox.io/s/confident-keller-gh0nc?file=/pages/index.js
